### PR TITLE
[Android] Enable Java 8 features in Dolphin Java code base

### PR DESCRIPTION
**Pull request by mahdihijazi**
Android supports Java 8 partially, this add the support and replace all anonymous class with lambdas or method reference where applicable. No changes in logic.

[for more details on what features from Java 8 android supports](https://developer.android.com/studio/write/java8-support.html#supported_features) 

one step toward less ugly java code :)

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java [hunk 1]
```diff
@@ -685,25 +632,21 @@ private void chooseController() {
...
+                            getResources().getStringArray(R.array.controllersValues)[indexSelected]);
+                });
+		builder.setPositiveButton(getString(R.string.ok), (dialogInterface, i) ->
+		{
+            editor.apply();
+
+            mEmulationFragment.refreshInputOverlay();
+
+            Toast.makeText(getApplication(), R.string.emulation_controller_changed, Toast.LENGTH_SHORT).show();
+        });
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java]: The ending curly brace should be aligned with the starting curly brace (here and 8 lines up).

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/EmulationFragment.java [hunk 2]
```diff
@@ -327,14 +320,11 @@ private void runWithValidSurface()
...
 				{
-					@Override
-					public void run()
-					{
-						NativeLibrary.SurfaceChanged(mSurface);
-						NativeLibrary.Run(mGamePath);
-					}},
+                    NativeLibrary.SurfaceChanged(mSurface);
+                    NativeLibrary.Run(mGamePath);
+                },
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/EmulationFragment.java]: Same here.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameDatabase.java [hunk 3]
```diff
@@ -262,33 +262,29 @@ else if (!folder.exists())
...
-				// Pass the result cursor to the consumer.
-				subscriber.onNext(resultCursor);
-
-				// Tell the consumer we're done; it will unsubscribe implicitly.
-				subscriber.onCompleted();
-			}
-		});
+		return Observable.create(subscriber ->
+        {
+            Log.info("[GameDatabase] Reading games list...");
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/GameDatabase.java]: The indentation levels here are weird... Are some lines using tabs and some lines using spaces?

**Comment by BhaaLseN** [comment]
Less boilerplate is always nice, but there is a lot of indent mismatch and space/tab mixing going on. Could you please check and reformat those sections to match the style of the rest of the code in those files?

**Comment by mbc07** [comment]
Did you import the preferences file ```code-style-java.jar``` from Dolphin's repository in your IDE? AFAIK one of the things it's supposed to automatically take care of is matching indentation and spacing to our code design guidelines...

**Comment by mahdihijazi** [comment]
Thanks @mbc07  for point that out, I was trying to modify my settings after each PR but this is way easier :)

@JosJuice I hope I fixed all issues, let me know if I missed any, thanks!

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java [hunk 4]
```diff
@@ -575,61 +553,32 @@ private void toggleControls() {
...
+		builder.setNeutralButton(getString(R.string.emulation_toggle_all), (dialogInterface, i) -> mEmulationFragment.toggleInputOverlayVisibility());
+		builder.setPositiveButton(getString(R.string.ok), (dialogInterface, i) ->
+		{
+            editor.apply();
 
-				mEmulationFragment.refreshInputOverlay();
-			}
-		});
+            mEmulationFragment.refreshInputOverlay();
+        });
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java]: This one still looks a bit weird.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java [hunk 5]
```diff
@@ -330,34 +312,30 @@ private void toggleMenu()
...
+                            showScreenshot();
+                        }
+
+                        @Override
+                        public void onError()
+                        {
+                            finish();
+                        }
+                    });
+        });
```
Comment (BhaaLseN) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java]: Here's still an indent mismatch with the opening brace on line 316

**Comment by mahdihijazi** [comment]
I fixed the last couple of style issues

**Pull request merged**